### PR TITLE
Improve memory usage on large messages in the state transitions panel.

### PR DESCRIPTION
**Pull request by foxymiles**
**User-Facing Changes**
Improve memory usage on large messages in the state transitions panel.

**Description**
During playback the StateTransition panel accumulates currentFrame messages via the useMessagesByPath hook. This PR expands topic slicing to that hook so that we only accumulate the portions of messages we actually need.

### CODE REVIEW

File: packages/studio-base/src/PanelAPI/useMessagesByTopic.test.tsx [hunk 1]
```diff
@@ -34,6 +34,20 @@ describe("useMessagesByTopic", () => {
...
+      {
+        initialProps: { topics: [{ topic: "/foo", fields: ["a"] }], historySize: 1 },
+        wrapper: ({ children }) => (
+          <MockMessagePipelineProvider>{children}</MockMessagePipelineProvider>
+        ),
+      },
+    );
+
+    expect(result.current).toEqual({ "/foo": [] });
+  });
```
Comment (jtbandes) [inline packages/studio-base/src/PanelAPI/useMessagesByTopic.test.tsx:49]: What is this test testing? It would be nice to actually see a "sliced" object show up in the results to be sure it works

File: packages/studio-base/src/players/subscribePayloadFromMessagePath.ts [hunk 2]
```diff
@@ -28,12 +28,12 @@ export function subscribePayloadFromMessagePath(
...
   if (!firstField) {
-    return { topic: parsedPath.topicName, preloadType };
+    return { topic: parsedPath.topicName, preloadType: preloadType ?? "partial" };
   }
 
   return {
     topic: parsedPath.topicName,
-    preloadType,
+    preloadType: preloadType ?? "partial",
     fields: [firstField.name],
```
Comment (jtbandes) [inline packages/studio-base/src/players/subscribePayloadFromMessagePath.ts:37]: What's the reason for this change? Is there a problem with passing an `undefined` value downstream?

**Review by jtbandes — APPROVED** [review]

File: packages/studio-base/src/PanelAPI/useMessagesByTopic.test.tsx (hunk 1 above)
Comment (foxymiles) [reply packages/studio-base/src/PanelAPI/useMessagesByTopic.test.tsx:49]: The actual slicing happens at the datasource level so the responsibility of this hook is just to pass through the subscription correctly.

File: packages/studio-base/src/players/subscribePayloadFromMessagePath.ts (hunk 2 above)
Comment (foxymiles) [reply packages/studio-base/src/players/subscribePayloadFromMessagePath.ts:37]: We initially made `preloadType` optional just to limit the scope of the PR but I'd like to start taking steps towards making it required and explicit.
